integration-cli/docker_cli_build_test.go

```diff
@@ -1977,26 +1973,20 @@ func TestBuildCancelationKillsSleep(t *testing.T) {
 	}
 	defer ctx.Close()
 
-	var wg sync.WaitGroup
-	defer wg.Wait()
-
```

Comment:
I would be interested to hear the reasoning behind removing the waitgroup. My reasoning behind having it was to ensure that if the goroutine did not close for some reason due to future modifications to the code, it would block, and goroutine leakage could be detected. I use this pattern in this manner to specify that I really want this goroutine not to last longer than the calling scope, and if it does it should be detected by someone, because the caller will block at that wait.

```go
var wg sync.WaitGroup
defer wg.Wait()
wg.Add(1)
go func() { 
  defer wg.Done()
  ...
}()
```